src/test/java/by/itechart/api/controller/UserControllerIT.java

```diff
@@ -124,6 +127,27 @@ void createUser() throws Exception {
                 .andExpect(jsonPath("$.phone", is("1231231231")));
     }
 
+    @Test
+    void shouldThrowEmailDuplicationExceptionWhenUserWithEmailExists() throws Exception {
+        var createUserOne = getCreateUserDTO();
+        var createUserTwo = getCreateUserDTO();
+        this.mockMvc.perform(post("/users")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(createUserOne))
+                .with(httpBasic(ADMIN_USERNAME, ADMIN_AND_USER_PASSWORD))
+                .with(csrf()))
+                .andExpect(status().isCreated());
+        Exception resolvedException = this.mockMvc.perform(post("/users")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(createUserTwo))
+                .with(httpBasic(ADMIN_USERNAME, ADMIN_AND_USER_PASSWORD))
+                .with(csrf()))
+                .andExpect(status().isBadRequest())
+                .andReturn().getResolvedException();
+        assertTrue(resolvedException instanceof UserEmailDuplicationException);
+        assertEquals("User with this email is already exists", resolvedException.getMessage());
```

Comment:
Another way to validate error message is using `jsonPath` as you did in previous cases:
`.andExpect(jsonPath("$.message", is("User with this email is already exists")));`
IMO, this way of verification is more preferable since it doesn't know about implementation details and it's shouldn't. Here you are expecting to get `UserEmailDuplicationException`, and if you change implementation by replacing `UserEmailDuplicationException` with `InvalidInputDataException`, you have to change tests too